python/paddle/tensor/math.py

```diff
@@ -321,8 +321,12 @@ def add_(x, y, name=None):
     if in_dygraph_mode():
         op_type = 'elementwise_add_'
         axis = -1
-        return _elementwise_op_in_dygraph(
+        inplace_shape = x.shape
+        out = _elementwise_op_in_dygraph(
             x, y, axis=axis, op_name=op_type)
+        if inplace_shape != out.shape:
```

Comment:
The check should be down before the execution of the op.

Reply:
done